### PR TITLE
[Neo VM] optimize newstruct

### PR DESCRIPTION
**Pull request by Jim8y**
# Description

This pr optimize the performance of opcode `newstruct` by 15 -20%. 

Fixes #

## Type of change

- [x] Optimization (the change is only an optimization)
- [ ] Style (the change is only a code style for better maintenance or standard purpose)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

**Test Configuration**:

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Review by vncoelho — COMMENTED** [review]
Is not the Repeat slightly slower in some cases due to the overhead of using LINQ (Enumerable)?

Maybe it introduces some additional method calls and iterator logic. 

Maybe something like:

```cs
// Cache the reference counter to avoid multiple property accesses
var referenceCounter = engine.ReferenceCounter;
Struct result = new(referenceCounter);
if (result is ICollection<StackItem> collection)
    collection.EnsureCapacity(n);

for (var i = 0; i < n; i++)
    result.Add(StackItem.Null);

engine.Push(result);
```

**Comment by vncoelho** [comment]
Something like
```cs
var referenceCounter = engine.ReferenceCounter;
var nullArray = new StackItem[n];
for (var i = 0; i < n; i++)
    nullArray[i] = StackItem.Null;

Struct result = new(referenceCounter, nullArray);
engine.Push(result);
```

**Review by shargon — DISMISSED** [review]
It was different than array `NewArray` weird? good optimization

```
var nullArray = new StackItem[n];
Array.Fill(nullArray, StackItem.Null);
```

This will be faster, for both methods, array and struct, I prefer to avoid linq

**Review by AnnaShaleva — APPROVED** [review]

**Comment by shargon** [comment]
> It was different than array `NewArray` weird? good optimization
> 
> ```
> var nullArray = new StackItem[n];
> Array.Fill(nullArray, StackItem.Null);
> ```
> 
> This will be faster, for both methods, array and struct, I prefer to avoid linq

@Jim8y could you update both methods to use Array.Fill ?

File: src/Neo.VM/JumpTable/JumpTable.Compound.cs [hunk 1]
```diff
@@ -180,8 +182,9 @@ public virtual void NewArray_T(ExecutionEngine engine, Instruction instruction)
                 (byte)StackItemType.ByteString => ByteString.Empty,
                 _ => StackItem.Null
             };
-
-            engine.Push(new VMArray(engine.ReferenceCounter, Enumerable.Repeat(item, n)));
+            var nullArray = new StackItem[n];
+            Array.Fill(nullArray, item);
+            engine.Push(new VMArray(engine.ReferenceCounter, nullArray));
```
Comment (shargon) [inline src/Neo.VM/JumpTable/JumpTable.Compound.cs:187]:
```suggestion
            var itemArray = new StackItem[n];
            Array.Fill(itemArray, item);
            engine.Push(new VMArray(engine.ReferenceCounter, itemArray));
```
Can be != than null

**Review by vncoelho — APPROVED** [review]

**Pull request merged**